### DISCUSSION TITLE
[core] Does PMD have a security vulnerabilities about com.fasterxml.jackson.core:jackson-annotations:jar:2.3.1?

### DISCUSSION THREAD

**Discussion by cyw3** [pmd/pmd#3051] [Q&A] [answered]
I found that when building PMD source code, maven would download jar org.sonatype.plugins:nexus-staging-maven-plugin:1.6.8 and would download its dependency com.fasterxml.jackson.core:jackson-annotations:jar:2.3.1.
And we know com.fasterxml.jackson(below 2.8.10) has a deserialization vulnerability.

![image](https://user-images.githubusercontent.com/11549103/31987040-24b6b79a-b930-11e7-99a5-c41b9283802b.png)

you can see here: 
http://www.cnvd.org.cn/flaw/show/CNVD-2017-04483

**Comment by jsotuyod** [comment, marked as answer]
@cyw3 those are not PMD's dependencies, but part of the `nexus-staging-maven-plugin`, used to publish new releases to the central repository. PMD does not ship jackson.

The usage that plugin does of jackson is unknown to me (it *may* be safe), but should probably be discussed with the maintainers of that plugin at https://issues.sonatype.org/projects/NEXUS/issues

PMD itself does include an affected jackson in tests, through [Wiremock](http://wiremock.org/). However, this usage is safe (it only loads safe local jsons from file) and is not a feature we actually use ourselves (we only use wiremock to mock loading xml rulesets)

**Comment by cyw3** [comment]
Actually, I do not really get it. You mean PMD is safe?

**Comment by cyw3** [comment]
And how i can build the source code without nexus-staging-maven-plugin ?

**Comment by jsotuyod** [comment]
@cyw3 yes, PMD itself is safe, but `nexus-staging-maven-plugin` may be affected (unlikely).

The `nexus-staging-maven-plugin` is only used to publish a release to maven central by the PMD staff (`./mvnw deploy`). You can remove it completely from the pom on your local copy and you would never know the difference.

**Comment by cyw3** [comment]
@jsotuyod well. I can remove it completely from the pom on my local copy and I can go on building pmd source code. However, i occur another question below:
![image](https://user-images.githubusercontent.com/11549103/32405077-40644170-c12c-11e7-9d5a-f8898c18365d.png)
and jetty also has a security vulnerabilitie. you can see with this url:
http://www.cvedetails.com/cve/CVE-2016-4800

now, i still can not build the pmd source code.

**Comment by jsotuyod** [comment]
@cyw3 according to that CVE report, the issue affects Jetty 9.3.x up to 9.3.9. This is Jetty 6.1.26, which is unaffected.

Also, once again, this is a test dependency, only used during tests execution, not part of the final build.
